crates/sui/src/client_ptb/parser.rs

```diff
@@ -48,6 +48,8 @@ struct ProgramParsingState {
     gas_budget: Option<Spanned<u64>>,
     gas_price: Option<Spanned<u64>>,
     gas_sponsor: Option<Spanned<NumericalAddress>>,
+    sender_set: bool,
```

Comment:
What do we use this bool for? Could we just have an optional `sender` metadata field?